gson/src/main/java/com/google/gson/internal/bind/TypeAdapters.java

```diff
@@ -776,9 +777,14 @@ public void write(JsonWriter out, Locale value) throws IOException {
 
     public EnumTypeAdapter(Class<T> classOfT) {
       try {
-        for (T constant : classOfT.getEnumConstants()) {
+        for (Field field : classOfT.getDeclaredFields()) {
+          if (!field.isEnumConstant()) {
+            continue;
+          }
+          field.setAccessible(true);
+          T constant = (T)(field.get(null));
```

Comment:
Please add a `@SuppressWarnings("unchecked")` for this variable.

Reply:
Fixed at ddee26e283d088fea1ec2807dfb258e9f36b2497